### PR TITLE
Add analysis of AVS3 coded video

### PR DESCRIPTION
**Pull request by paulhiggs**
This PR adds support for AVS3 coded video using the existing AvsV as a base. Decoding of meaningful syntax elements from the sequence header including HDR Dynamic Metadata according to DVB Bluebook A001.

Commit f809737 updates the project libraries (I only have vc17 on my build system)

### CODE REVIEW

**Comment by JeromeMartinez** [comment]
Great PR!
Could you share at least 1 sample file?

File: Project/MSVC2022/Library/MediaInfoLib.vcxproj [hunk 1]
```diff
@@ -276,6 +276,7 @@
     <ClCompile Include="..\..\..\Source\MediaInfo\Text\File_Ttml.cpp" />
     <ClCompile Include="..\..\..\Source\MediaInfo\TimeCode.cpp" />
     <ClCompile Include="..\..\..\Source\MediaInfo\Video\File_Av1.cpp" />
+    <ClCompile Include="..\..\..\Source\MediaInfo\Video\File_Avs3.cpp" />
```
Comment (JeromeMartinez) [inline Project/MSVC2022/Library/MediaInfoLib.vcxproj:279]: Please same for all MSVC\* directory (stupid copy/pastes to do but we currently have different directory per MSVC version).

**Comment by paulhiggs** [comment]
This transport stream file ([#1-avs3-str-QP30.ts](https://www.dropbox.com/s/papyxuwwrxqtavs/%231-avs3-str-QP30.ts?dl=0)) contains AVS3 SDR video at 1920x1080 resolution. I have some other files in development that have additional features.

**Comment by JeromeMartinez** [comment]
> I have some other files in development that have additional features.

Would be great that all code paths are with a corresponding file, easier for automatic regression tests.

File: Project/MSVC2022/Library/MediaInfoLib.vcxproj (hunk 1 above)
Comment (JeromeMartinez) [reply Project/MSVC2022/Library/MediaInfoLib.vcxproj:279]: \*.filters need to be adapted too.

Comment (JeromeMartinez) [reply Project/MSVC2022/Library/MediaInfoLib.vcxproj:279]: Actually the other projects (Automake etc) are also not updated, I'll do it, faster.

**Comment by JeromeMartinez** [comment]
AVS3 seems not too different compared to AVS1/2, is it too difficult to expand `File_AvsV.cpp` rather than having another source file with lot of code duplicated?

**Comment by paulhiggs** [comment]
There are certainly some aspects which are similar, but AVS3 has a much extended toolset and syntax.
Same is true in part for VVC which has HEVC as a base with addition tools, options and syntax
________________________________

**Comment by JeromeMartinez** [comment]
> There are certainly some aspects which are similar, but AVS3 has a much extended toolset and syntax.

Extension of a format does not require to duplicate code, e.g. MPEG-2 Video uses MPEG-1 Video code and extends it.
I'll check a bit more the code next week and may suggest changes depending on the similarity (or non similarity) of bitstreams.

> Same is true in part for VVC which has HEVC as a base with addition tools, options and syntax

No VVC yet in MediaInfo ;-p, feel free to add support, it would be appreciated.

**Comment by paulhiggs** [comment]
Here are two other sample files developed in the DVB Validation and Verification activity

- [DVB_AVS3_VaV_No2-r3_2160P50_PQ10_Three_DMI.ts](https://www.dropbox.com/s/3nevljkuuwzm310/DVB_AVS3_VaV_No2-r3_2160P50_PQ10_Three_DMI.ts?dl=0)  - AVS3 4K with HDR10+, Dolby Vision and SL-HDR2 dynamic metadata
- [DVB_AVS3_VaV_No12-r4_4320P100_HLG10_HFR.ts](https://www.dropbox.com/s/76niy0n9uiftu8a/DVB_AVS3_VaV_No12-r4_4320P100_HLG10_HFR.ts?dl=0) - 8K AVS3 HFR with HLG10

**Comment by JeromeMartinez** [comment]
Please let this PR open, we are definitely super later on merging it but we are interested in it.

**Comment by JeromeMartinez** [comment]
> Here are two other sample files developed in the DVB Validation and Verification activity

Please push again the files.

**Comment by paulhiggs** [comment]
All AVS3 Validation and Verification biutstreams are now available in the [DVB Website](https://dvb.org/specifications/verification-validation/)

**Pull request closed without merging**